nx/lib/nx.ex

```diff
@@ -5192,6 +5192,33 @@ defmodule Nx do
     end
   end
 
+  @doc """
+  Calculates the complex conjugate of each element in the tensor.
+
+  If $z = a + bi$, $conjugate(z) = z^* = a - bi$
+
+  ## Examples
+
+       iex>Nx.conjugate(Complex.new(1, 2))
+       #Nx.Tensor<
+         c64
+         1.0-2.0i
+       >
+
+
+       iex> Nx.conjugate(Nx.tensor([Complex.new(1, 2), Complex.new(2, -4)]))
+       #Nx.Tensor<
+         c64[2]
+         [1.0-2.0i, 2.0+4.0i]
+       >
+  """
+  @doc type: :element
+  def conjugate(tensor) do
+    tensor = to_tensor(tensor)
+
+    impl!(tensor).conjugate(tensor, tensor)
```

Comment:
There's an issue with the output type here.

Try a test/example in which the input is s64 ou f32. I'd expect the output type to be inferred incorrectly (it should be the corresponding complex type).

I think we need a `to_complex` function in Nx.Types, which promotes everything below f32 to c64, f64 -> c128 and keeps complex types as the same.

Then you would have somthing like:
```suggestion
    impl!(tensor).conjugate(%{tensor | type: Nx.Types.to_complex(tensor.type)}, tensor)
```

Reply:
Regarding EXLA/Torchx, we can go ahead and merge this after this adjustment :)

Reply:
Oh, and regarding the failing torchx tests, there's a list to which you can add the tests in torchx/test/nx_doctest_test.exs

Reply:
Ah, yes I forgot about it :)